apps/meteor/client/views/audit/components/SecurityLogsTable.tsx

```diff
@@ -0,0 +1,211 @@
+		},
+	});
+
+	return (
+		<>
+			<Box mb={16} display='flex' flexWrap='nowrap' alignItems='flex-end'>
+				<Field flexShrink={1}>
+					<Margins inline={6}>
+						<FieldLabel>{t('Date')}</FieldLabel>
+						<DateRangePicker display='flex' flexGrow={1} value={dateRange} onChange={setDateRange} />
+					</Margins>
+				</Field>
+				<Field width={300}>
+					<Margins inline={6}>
+						<FieldLabel>{t('Setting')}</FieldLabel>
+						<SettingSelect value={setting} onChange={setSetting} />
+					</Margins>
+				</Field>
+				<ButtonGroup>
+					<Margins inline={6}>
+						<Button secondary onClick={handleClearFilters}>
+							{t('Clear_filters')}
+						</Button>
+						<Button primary onClick={handleApplyFilters}>
+							{t('Apply_filters')}
+						</Button>
+					</Margins>
+				</ButtonGroup>
+			</Box>
+
+			{isLoading && (
+				<GenericTable>
+					<GenericTableHeader>
+						<GenericTableHeaderCell>{t('Actor')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Timestamp')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Setting')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Changed_from')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Changed_to')}</GenericTableHeaderCell>
+					</GenericTableHeader>
+					<GenericTableBody>
+						<GenericTableLoadingRow cols={5} />
+					</GenericTableBody>
+				</GenericTable>
+			)}
+			{isSuccess && data.total === 0 && (
+				<GenericNoResults
+					title={t('No_results_found')}
+					description={t('Try_different_filters')}
+					buttonTitle={t('Clear_filters')}
+					buttonAction={handleClearFilters}
+				/>
+			)}
+			{isSuccess && data.total > 0 && (
+				<GenericTable>
+					<GenericTableHeader>
+						<GenericTableHeaderCell>{t('Actor')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Timestamp')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Setting')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Changed_from')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Changed_to')}</GenericTableHeaderCell>
+					</GenericTableHeader>
+					<GenericTableBody>
+						{data.events.map((item) => (
+							<GenericTableRow
+								key={item.ts}
+								role='link'
+								action
+								tabIndex={0}
+								onClick={() =>
+									handleItemClick({
+										actor: item.actor.type === 'user' ? item.actor.username : item.actor.type,
+										timestamp: new Date(item.ts).toDateString(),
+										setting: item.data[0].value,
+										changedFrom: String(item.data[1].value),
+										changedTo: String(item.data[2].value),
+									})
+								}
+							>
+								<GenericTableCell withTruncatedText>
+									<Box display='flex' alignItems='center'>
+										{item.actor.type === 'user' && (
+											<Box mie={4}>
+												<UserAvatar size='x24' username={item.actor.username} />
```

Comment:
The setting audit record does not update when usernames change, so you need to use the avatar with the user id instead of username.